pkg/actuators/machine/reconciler.go

```diff
@@ -446,3 +474,226 @@ func (r *Reconciler) setMachineAddresses(instance *models.PVMInstance) error {
+	// construct the map from provided LoadBalancer names for easy search
+	loadBalancerNamesMap := make(map[string]struct{})
+	for _, lb := range loadBalancerNames {
+		loadBalancerNamesMap[lb] = struct{}{}
+	}
+
+	listLoadBalancersOptions := &vpcv1.ListLoadBalancersOptions{}
+	loadBalancersList, _, err := powerVSClient.ListLoadBalancers(listLoadBalancersOptions)
+	if err != nil {
+		return nil, err
+	}
+	if loadBalancersList == nil {
+		return nil, fmt.Errorf("no loadbalancer is retrieved")
+	}
+
+	// construct LoadBalancer name with LoadBalancer ID map
+	lbMap := make(map[string]string)
+	for _, lb := range loadBalancersList.LoadBalancers {
+		if _, ok := loadBalancerNamesMap[*lb.Name]; ok {
+			lbMap[*lb.Name] = *lb.ID
+		}
+	}
+	if len(lbMap) != len(loadBalancerNames) {
+		return nil, fmt.Errorf("not able to find %s loadbalancer in cloud", loadBalancerNames)
+	}
+	return lbMap, nil
+}
+
+func createLoadBalancerPoolMember(powerVSClient client.Client, lbMap map[string]string, internalIP string) error {
+	// update LoadBalancer pool for each LoadBalancer
+	for lbName, lbID := range lbMap {
+		// Fetch the LoadBalancer details
+		loadBalancer, _, err := powerVSClient.GetLoadBalancer(&vpcv1.GetLoadBalancerOptions{
+			ID: &lbID,
+		})
+		if err != nil {
+			return err
+		}
+		if len(loadBalancer.Pools) == 0 {
+			return fmt.Errorf("no pools exist for the load balancer")
+		}
+		if *loadBalancer.ProvisioningStatus != "active" {
+			klog.Infof("not able to update pool for loadbalancer %s, load balancer is not in active state", lbName)
+			return &machinecontroller.RequeueAfterError{RequeueAfter: requeueAfterSeconds * time.Second}
+		}
+
+		// Update each LoadBalancer pool
+		for _, pool := range loadBalancer.Pools {
+			// TODO: (question): will bootstrap-node pool exist or will it be cleaned up by installer?
+			if *pool.Name == "bootstrap-node" {
+				continue
+			}
+			klog.Infof("Updating LoadBalancer pool member %s for LoadBalancer %s with IP %s", *pool.Name, lbName, internalIP)
+			listOptions := &vpcv1.ListLoadBalancerPoolMembersOptions{}
+			listOptions.SetLoadBalancerID(*loadBalancer.ID)
+			listOptions.SetPoolID(*pool.ID)
+			listLoadBalancerPoolMembers, _, err := powerVSClient.ListLoadBalancerPoolMembers(listOptions)
+			if err != nil {
+				return errors.Wrapf(err, "failed to list %s LoadBalancer pool %v", *pool.Name, err)
+			}
+			var targetPort int64
+			var alreadyRegistered bool
+			for _, member := range listLoadBalancerPoolMembers.Members {
+				if target, ok := member.Target.(*vpcv1.LoadBalancerPoolMemberTarget); ok {
+					targetPort = *member.Port
+					if *target.Address == internalIP {
+						alreadyRegistered = true
+						klog.Infof("Target with IP %s already configured for pool %s", internalIP, *pool.Name)
+						continue
+					}
+				}
+			}
+			if !alreadyRegistered {
+				// make sure that LoadBalancer is in active state
+				loadBalancer, _, err := powerVSClient.GetLoadBalancer(&vpcv1.GetLoadBalancerOptions{
```

Comment:
Do we need another Get call here?

Reply:
In our case a loadbalancer will have 2 pools, So after updating one pool, if we try to update second one it will fail as the loadbalancer wont be in active state, so instead of throwing ugly error added a check.